cllc-public-app/ClientApp/src/app/shared/components/related-licence-picker/related-licence-picker.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit, ViewChild, Output, EventEmitter } from "@angular/core";
+import { MatAutocompleteTrigger } from "@angular/material/autocomplete";
+import { FormBuilder, FormGroup } from "@angular/forms";
+import { AccountDataService } from "@services/account-data.service";
+import { filter, tap, switchMap } from "rxjs/operators";
+import { LicenseDataService } from "@services/license-data.service";
+import { RelatedLicence } from "@models/related-licence";
+
+@Component({
+  selector: 'app-related-licence-picker',
+  templateUrl: './related-licence-picker.component.html',
+  styleUrls: ['./related-licence-picker.component.scss']
+})
+export class RelatedLicencePickerComponent implements OnInit {
+  @ViewChild("autocomplete", { read: MatAutocompleteTrigger, static: true })
+  inputAutoComplete: MatAutocompleteTrigger;
+  @Output()
+  valueSelected = new EventEmitter<string>();
+  form: FormGroup;
+  autocompleteLicences: any[];
+  licenceRequestInProgress: boolean;
+
+
+  constructor(private licenceDataService: LicenseDataService,
+    private fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.form = this.fb.group({
+      autocompleteInput: [""]
+    });
+
+    this.form.get("autocompleteInput").valueChanges
+      .pipe(filter(value => value && value.length >= 3),
+        tap(_ => {
+          this.autocompleteLicences = [];
+          this.licenceRequestInProgress = true;
+        }),
+        switchMap(value => this.licenceDataService.getAutocomplete(value))
+      )
+      .subscribe(data => {        
+        this.autocompleteLicences = data;
+        this.licenceRequestInProgress = false;
+        this.inputAutoComplete.openPanel();
+      });
+  }
```

Comment:
This looks great. I'd suggest looking into `debounce()` - [link](https://rxjs-dev.firebaseapp.com/api/operators/debounce) to help with performance. No need to fire off API requests as the user is typing into the auto-complete box. We can discuss further tomorrow the way I've seen it done in other projects.

Reply:
Good idea.  I suggest we save that for a future update though